sycl/test/devicelib/separate_compile_test.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clangxx -fsycl -fsycl-link %S/std_complex_math_test.cpp -o %t_device.o
+// RUN: %clangxx -fsycl-device-only -Xclang -fsycl-int-header=std_complex_math_test_ihdr.h %S/std_complex_math_test.cpp -I %sycl_include -Wno-sycl-strict
+// >> host compilation...
+// RUN: %clangxx -include std_complex_math_test_ihdr.h -c %S/std_complex_math_test.cpp -o %t_host.o -I %sycl_include -Wno-sycl-strict
+// RUN: %clangxx %t_host.o %t_device.o -o %t.out -lsycl
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// RUN: %clangxx -fsycl -fsycl-link -fsycl-device-lib=all %S/std_complex_math_fp64_test.cpp -o %t_fp64_device.o
+// RUN: %clangxx -fsycl-device-only -Xclang -fsycl-int-header=std_complex_math_fp64_test_ihdr.h %S/std_complex_math_fp64_test.cpp -I %sycl_include -Wno-sycl-strict
+// >> host compilation...
+// RUN: %clangxx -include std_complex_math_fp64_test_ihdr.h -c %S/std_complex_math_fp64_test.cpp -o %t_fp64_host.o -I %sycl_include -Wno-sycl-strict
+// RUN: %clangxx %t_fp64_host.o %t_fp64_device.o -o %t_fp64.out -lsycl
+// RUN: %CPU_RUN_PLACEHOLDER %t_fp64.out
+// RUN: %ACC_RUN_PLACEHOLDER %t_fp64.out
```

Comment:
I'm only curious: are there any other use-cases?

Reply:
Hi, @s-kanaev 
Do you mean are there any other use cases for device libraries?

Reply:
Yes, exactly.

Reply:
We have supported for normal jit and aot including fintelfpga mode. However, there are many other use cases which is not so "popular", I believe we do have some use cases which have not been handled, for example: it seems that "-fsycl-link-targets=<T1,...,Tn>" also needs the support.
We need to review all options in https://github.com/intel/llvm/blob/sycl/sycl/doc/UsersManual.md to see if anything missed.
Thanks very much.